cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +353,24 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+	k8sVersion, err := version.ParseSemantic(kubever.KubernetesVersion)
+	if err != nil {
+		return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+	}
+
+	kubeadmVersion := apiversion.Get()
```

Comment:
There is a special case to take into account here, this version may be `v0.0.0` as well (in `pkg/version` on all unreleased versions). In that case, we should just skip this check, as the user has built kubeadm himself/herself in a custom manner and really knows what he/she is doing. 

When compiling kubeadm with `make WHAT=cmd/kubeadm` the version is set properly, even when compiling from master, it's only the super-custom build case that has `v0.0.0`